### PR TITLE
Event widget

### PR DESCRIPTION
**Pull request by joanniclaborde**
There are still 12 `TODO` in the code, mainly:
 - [ ] Signup links for exclusive events
 - [x]  events_to_json
 - [ ]  RSVP and comments tallies
 - [x]  Data not imported in Universe, see https://github.com/SparkartGroupInc/sparkart-tools/issues/479#issuecomment-73773619
 - [ ]  Event passwords/codes for signed in users
 - [x]  browse_program

### CODE REVIEW

File: widgets/event/browse_compact.hbs [hunk 1]
```diff
@@ -0,0 +1,81 @@
+{{! TODO
+<script type="application/ld+json">
+  <%= events_to_json(events, options) %>
+</script>
```
Comment (pushred) [inline widgets/event/browse_compact.hbs]: See https://github.com/SparkartGroupInc/keithurban.net/blob/master/preprocessors/add_tour_metadata.js

File: widgets/utils/universe.js [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  if (pages.current_page > 1) result.push({page: pages.current_page - 1, class: 'back', label: 'Back'});
+  if (pages.current_page < pages.total_pages) result.push({page: pages.current_page + 1, class: 'next', label: 'Next'});
+
+  return result;
+};
+
+module.exports.eventEnded = function(date) {
+  if (!date) return false;
+
+  return new Date > (new Date(date)).setHours(23, 59, 59, 999);
```
Comment (pushred) [inline widgets/utils/universe.js]: I didn't know you could compare date objects like this!

Comment (joanniclaborde) [inline widgets/utils/universe.js]: The internet told me.

**Comment by pushred** [comment]
I did some sanity checks of comparisons with native date objects, this should be fine since it'll only be running in node.

Re: the naming of the eventEnded helpers, we should follow the `isBoolean` convention for this since we can't use `?` in Handlebars. Also should keep it loose for use with any date, there's nothing unique about events. So `isEnded`.

**Comment by pushred** [comment]
Also generally.. why all the event-specific helpers? Is the logic that bad?

**Comment by joanniclaborde** [comment]
The `.setHours(23, 59, 59, 999)` part is unique to events: an event is considered ended when the whole day is over. I can rename it to `isEventEnded` though.

Yes, that widget is horrible...

File: widgets/utils/universe.js [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    }
+    return memo;
+  }, []);
+};
+
+var sold_out = function(event) {
+  return event.links.length > 0 && !_.find(event.links, function(link) {return link.status !== 'Sold Out'});
+};
+
+var tagsTokind = function(tags) {
```
Comment (pushred) [inline widgets/utils/universe.js]: Shouldn't that be `tagsToKind`?

File: widgets/utils/universe.js [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+      memo.push({
+        tag:   tag.replace(/ /g, '-'),
+        title: tag
+      });
+    }
+    return memo;
+  }, []);
+};
+
+var sold_out = function(event) {
```
Comment (pushred) [inline widgets/utils/universe.js]: `isSoldOut`

Comment (joanniclaborde) [inline widgets/utils/universe.js]: Old Ruby habit...

**Pull request merged**